cdap-app-fabric/src/main/java/io/cdap/cdap/scheduler/Scheduler.java

```diff
@@ -138,6 +138,15 @@ void addSchedules(Iterable<? extends ProgramSchedule> schedules)
    */
   ProgramSchedule getSchedule(ScheduleId scheduleId) throws NotFoundException;
```

Comment:
unless this is cheaper to get (1 I/O vs multiple I/O or something like that), it seems like it should be replaced.

Reply:
Yes. Plan to deprecate it and replace with the new getScheduleRecord. 

Added a TODO to clean up in a follow-up change